Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -578,24 +622,41 @@ class TrackStateProxy {
   /// Full calibrated measurement vector. Might contain additional zeroed
   /// dimensions.
   /// @return The measurement vector
-  Measurement calibrated() const;
+  template <size_t measdim>
+  Measurement<measdim> calibrated() const;
 
   /// Full calibrated measurement covariance matrix. The effective covariance
   /// is located in the top left corner, everything else is zeroed.
   /// @return The measurement covariance matrix
-  MeasurementCovariance calibratedCovariance() const;
+  template <size_t measdim>
+  MeasurementCovariance<measdim> calibratedCovariance() const;
 
   /// Dynamic measurement vector with only the valid dimensions.
   /// @return The effective calibrated measurement vector
   auto effectiveCalibrated() const {
-    return calibrated().head(calibratedSize());
+    double* data{nullptr};
+
+    visit_measurement(*this, calibratedSize(), [&](auto meas, auto cov) {
+      (void)cov;
+      data = const_cast<double*>(meas.data());
```

Comment:
Can you comment on the `const_cast` here? I quickly tried to look through this, but didn't really understood why this is necessary here. Is it because the `Eigen` types require non-const pointers? And are we sure that we don't invoke UB here?

Maybe i would be good to add an extra comment on why this is used...

Reply:
should be gone i think